src/tests/unit/tests/background/stores/card-selection-store.test.ts

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { RuleExpandCollapsePayload, UnifiedScanCompletedPayload } from '../../../../../background/actions/action-payloads';
+import { CardSelectionActions } from '../../../../../background/actions/card-selection-actions';
+import { UnifiedScanResultActions } from '../../../../../background/actions/unified-scan-result-actions';
+import { CardSelectionStore } from '../../../../../background/stores/card-selection-store';
+import { StoreNames } from '../../../../../common/stores/store-names';
+import { CardSelectionStoreData } from '../../../../../common/types/store-data/card-selection-store-data';
+import { UnifiedResult } from '../../../../../common/types/store-data/unified-data-interface';
+import { createStoreWithNullParams, StoreTester } from '../../../common/store-tester';
+
+describe('CardSelectionStore Test', () => {
+    test('constructor has no side effects', () => {
+        const testObject = createStoreWithNullParams(CardSelectionStore);
+        expect(testObject).toBeDefined();
+    });
+
+    test('getId', () => {
+        const testObject = createStoreWithNullParams(CardSelectionStore);
+
+        expect(testObject.getId()).toEqual(StoreNames[StoreNames.CardSelectionStore]);
+    });
+
+    test('check defaultState is as expected', () => {
+        const defaultState = getDefaultState();
+
+        expect(defaultState.rules).toBeDefined();
+    });
+
+    test('onScanCompleted', () => {
+        const initialState = getDefaultState();
+
+        const payload: UnifiedScanCompletedPayload = {
+            scanResult: [
+                {
+                    ruleId: 'sampleRuleId',
+                    uid: 'sampleUid',
+                    status: 'fail',
+                } as UnifiedResult,
+            ],
+            rules: [],
+        } as UnifiedScanCompletedPayload;
+
+        const expectedState: CardSelectionStoreData = {
+            rules: {
+                sampleRuleId: {
+                    isExpanded: false,
+                    cards: {
+                        sampleUid: false,
+                    },
+                },
+            },
+        };
+
+        createStoreForUnifiedScanResultActions('scanCompleted')
+            .withActionParam(payload)
+            .testListenerToBeCalledOnce(initialState, expectedState);
+    });
+
+    test('ToggleRuleExpandCollapse expanded', () => {
+        const initialState: CardSelectionStoreData = {
```

Comment:
I'm seeing some duplicate state declarations. Up to you but I think you could maybe create variables in a beforeEach loop instead to minimize that or maybe have a ```getState``` function which adjusts the properties accordingly to it's params. It should make the tests less verbose/remove duplication.

Reply:
Fixed.